BUILDING.md

```diff
@@ -0,0 +1,71 @@
+# Building with CMake
+
+## vcpkg dependencies
+
+This project uses vcpkg to manage its dependencies, but this is optional. You
+may also install those dependencies manually or using any other package
+manager.
+
+If you have vcpkg set up, then you have to provide some extra flags to the
+configure command, which differs only in how you provide the paths depending on
+your OS:
+
+```batch
+rem Windows
+-D "CMAKE_TOOLCHAIN_FILE=%VCPKG_ROOT%/scripts/buildsystems/vcpkg.cmake"
+```
+
+```sh
+# Unix based (Linux, macOS)
+-D "CMAKE_TOOLCHAIN_FILE=$VCPKG_ROOT/scripts/buildsystems/vcpkg.cmake"
+```
+
+You may also set the `VCPKG_MANIFEST_FEATURES` variable to `magic-enum` if you
+would like to enable support for it.
```

Comment:
Cool, I installed vcpkg.
Configured with

`cmake -S . -B build -D CMAKE_BUILD_TYPE=Release -DCMAKE_TOOLCHAIN_FILE="$VCPKG_ROOT/scripts/buildsystems/vcpkg.cmake" -Denum_set_DEVELOPER_MODE=ON -DVCPKG_MANIFEST_FEATURES=magic-enum`

and try to build with

`cmake --build build`

but it can not find the magic enum header, do we need to set include path through cmake manually?

Reply:
The BUILDING document is for consumers of the library, not the developers. Everything is set up in terms of CMake.

Reply:
Hmm it works through docker, must be my local setup that is just too old.